### DISCUSSION TITLE
반려 식물 단건 조회 시 본인만 조회 가능하도록 제약을 걸어야하는가?

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#166] [뜨거운 토론]
## 논제

반려 식물의 CRUD를 수행할 때 사용자의 인증정보를 받아오고있습니다.
Update, Delete에 대해서는 본인에 대한 검증이 필요하지만 반려 식물 조회에서도 검증을 넣어야할지 약간 긴가민가해서 여러분의 의견을 듣고싶습니다.

1. 반려 식물 단건조회 시 본인의 정보가 아니면 예외를 던진다.
2. 반려 식물 단건 조회 시 본인의 정보가 아니여도 조회할 수 있게 한다.

저는 개인적으로 1번 의견입니다.

현재 피움의 서비스 구성 상 다른 사용자의 반려 식물을 조회하는 페이지 혹은 기능이 존재하지 않습니다.
때문에 다른 사용자의 반려 식물에 대한 단건조회는 의도되지 않은 동작이라고 생각하여 검증을 수행해야한다고 생각하기 때문입니다.

여러분의 의견이 궁금합니다

**Comment by WaiNaat** [comment]
저는 1번이요
본인의 반려 식물 정보도 사생활이라 생각합니다

**Comment by yeonkkk** [comment]
저도 1번이요!
만약 우리 서비스에서 커뮤니티성 기능이 있다면 고려해볼만 하지만,  지금은 개인의 관리 정보만 기록하고 있기 때문에 오픈하지 않는게 좋을 것 같습니다.

**Comment by Kim0914** [comment]
현재 저희 서비스에서는 반려 식물을 조회하는 기능이 **로그인을 한 사용자 본인만 확인할 수 있는 기능**이기 때문에 체크해줘야할 것 같네요 🙃

**Comment by Choi-JJunho** [comment]
그렇다면 `반려 식물 단건 히스토리 조회 API`에 대해서도 동일하게 검증을 수행해야겠군요~

**Comment by Choi-JJunho** [comment]
## 결론

반려 식물과 연관된 기능에 대한 논의를 거쳤다.
현재 기획 상 반려식물과 연관된 정보는 로그인을 해야만 볼 수 있는 정보이다.
또한 커뮤니티 기능이 없기 때문에 사용자 개개인의 정보는 개인화 되어있어야 한다. 
이에 다른 사용자의 반려 식물 정보를 조회하는 행위에 대해 검증을 수행해야한다.

현재 개발단계에서 조회 시 고려할 사항은 다음과 같다

- 반려 식물 히스토리 단건 조회
- 반려 식물 단건 조회
- 반려 식물 정보 수정 등

위와 같은 결론으로 마무리지어져 close하겠습니다~
